src/plugins/explore/public/components/query_panel/components/footer/recent_query/table.tsx

```diff
@@ -0,0 +1,158 @@
+        query: 'source=metrics | where error=500',
+        language: LanguageType.KeyValue,
+        prompt: 'Give me the average CPU usage by host',
+      },
+      timeRange: { from: 'now-24h', to: 'now' },
+      time: Date.now() - 86400000,
+    },
+  ];
+
+  // TODO: dispatch action to get recent queries.
+  // const [recentQueries, setRecentQueries] = useState<RecentQueryItem[]>(mockRecentQueries);
+
+  const getRowProps = (item: RecentQueryTableItem) => ({
+    'data-test-subj': `row-${item.id}`,
+    className: 'customRowClass',
+    onClick: () => {},
+  });
+
+  const getCellProps = (
+    item: RecentQueryTableItem,
+    column: EuiBasicTableColumn<RecentQueryTableItem>
+  ) => ({
+    className: 'customCellClass',
+    'data-test-subj': `cell-${item.id}-${(column as { field?: string }).field ?? 'actions'}`,
+    textOnly: true,
+  });
+
+  const tableColumns: Array<EuiBasicTableColumn<RecentQueryTableItem>> = [
+    {
+      field: 'query',
+      name: i18n.translate('explore.queryPanel.recentQueryTable.queryColumn', {
+        defaultMessage: 'Recent query',
+      }),
+    },
+    {
+      field: 'time',
+      name: i18n.translate('explore.queryPanel.recentQueryTable.lastRunColumn', {
+        defaultMessage: 'Last run',
+      }),
+    },
+    {
+      name: i18n.translate('explore.queryPanel.recentQueryTable.actionsColumn', {
+        defaultMessage: 'Actions',
+      }),
+      actions: [
+        {
+          name: i18n.translate('explore.queryPanel.recentQueryTable.runAction', {
+            defaultMessage: 'Run',
+          }),
+          description: i18n.translate('explore.queryPanel.recentQueryTable.runActionDescription', {
+            defaultMessage: 'Run recent query',
+          }),
+          icon: 'play',
+          type: 'icon',
+          onClick: (item: RecentQueryTableItem) => {
+            const foundQuery = recentQueries.find((recentQuery) => recentQuery.id === item.id);
+            if (foundQuery) {
+              onClickRecentQuery(foundQuery.query, foundQuery.timeRange);
+            }
+          },
+          'data-test-subj': 'action-run',
+        },
+        {
+          render: (item: RecentQueryTableItem) => (
+            <EuiCopy textToCopy={item.query as string}>
+              {(copy) => (
+                <EuiButtonIcon
+                  onClick={copy}
+                  iconType="copyClipboard"
+                  aria-label={i18n.translate(
+                    'explore.queryPanel.recentQueryTable.copyActionAriaLabel',
+                    {
+                      defaultMessage: 'Copy recent query',
+                    }
+                  )}
+                />
+              )}
+            </EuiCopy>
+          ),
+        },
+      ],
+      width: '70px',
+    },
+  ];
+
+  const recentQueryItems: RecentQueryTableItem[] = recentQueries
+    .filter((item, idx) => idx < MAX_RECENT_QUERY_SIZE)
+    .map((query) => ({
+      id: query.id,
+      query: query.query.query,
```

Comment:
do they all have to be called query? i know there are types but it's hard to tell which level means what by looking at the code